File: src/main/java/com/sudosoo/takeiteasy/service/CategoryService.java
package com.sudosoo.takeiteasy.service;

import com.sudosoo.takeiteasy.dto.category.CategoryResponseDto;
import com.sudosoo.takeiteasy.dto.category.CreateCategoryRequestDto;
import com.sudosoo.takeiteasy.entity.Category;
import org.springframework.data.domain.Pageable;


public interface CategoryService {
    Category createCategory(CreateCategoryRequestDto createCategoryRequestDto);
    Category getCategoryByCategoryId(Long categoryId);
    CategoryResponseDto getPostsByCategoryId(Long categoryId, Pageable pageable);
}

File: takeItEasy-application/src/main/kotlin/com/sudosoo/takeItEasy/service/CategoryService.kt
package com.sudosoo.takeItEasy.service

import com.sudosoo.takeItEasy.dto.category.CategoryResponseDto
import com.sudosoo.takeItEasy.dto.category.CreateCategoryRequestDto
import com.sudosoo.takeiteasy.entity.Category
import org.springframework.data.domain.Pageable

interface CategoryService {
    fun createCategory(createCategoryRequestDto: CreateCategoryRequestDto): Category
    fun getById(categoryId: Long): Category
    fun getPostsByCategoryId(categoryId: Long, pageable: Pageable): CategoryResponseDto
}
